From 13108eb57c73e95be902f4b8b3994b6e7f5e0b8e Mon Sep 17 00:00:00 2001
From: Vinith Kumar R <quic_vinithku@quicinc.com>
Date: Tue, 8 Oct 2024 11:44:16 +0530
Subject: [PATCH] wifi: ath12k: initialize local variable to avoid exception

Currently, the local variables such as band, hdr and queue are
uninitialized in the function definition. But in some cases there is
a chance of accessing these variables and leading to a crash or
exception. So, avoiding the same by initializing the value to 0

Signed-off-by: Vinith Kumar R <quic_vinithku@quicinc.com>
---
 net/mac80211/tx.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/net/mac80211/tx.c b/net/mac80211/tx.c
index 56f5471..56f6e63 100644
--- a/net/mac80211/tx.c
+++ b/net/mac80211/tx.c
@@ -2654,7 +2654,7 @@ static struct sk_buff *ieee80211_build_hdr(struct ieee80211_sub_if_data *sdata,
 	int head_need;
 	u16 ethertype, hdrlen,  meshhdrlen = 0;
 	__le16 fc;
-	struct ieee80211_hdr hdr;
+	struct ieee80211_hdr hdr = {0};
 	struct ieee80211s_hdr mesh_hdr __maybe_unused;
 	struct mesh_path __maybe_unused *mppath = NULL, *mpath = NULL;
 	const u8 *encaps_data;
@@ -2664,7 +2664,7 @@ static struct sk_buff *ieee80211_build_hdr(struct ieee80211_sub_if_data *sdata,
 	bool multicast;
 	u16 info_id = 0;
 	struct ieee80211_chanctx_conf *chanctx_conf = NULL;
-	enum nl80211_band band;
+	enum nl80211_band band = 0;
 	int ret;
 	u8 link_id = u32_get_bits(ctrl_flags, IEEE80211_TX_CTRL_MLO_LINK);
 
@@ -4771,7 +4771,7 @@ static void ieee80211_8023_xmit(struct ieee80211_sub_if_data *sdata,
 	struct tid_ampdu_tx *tid_tx;
 	struct sk_buff *seg, *next;
 	unsigned int skbs = 0, len = 0;
-	u16 queue;
+	u16 queue = 0;
 	unsigned char *ra = ehdr->h_dest;
 	bool multicast;
 	u8 tid;
-- 
2.34.1

